### PR TITLE
feat(cmd/influx): add org flag to bucket command

### LINKED ISSUE #10716: creating bucket using organization id leaves organization name empty
- Using `influx` tool create bucket using `--org-id` flag
> influx bucket create -n "bucket-influx-4" -r "1s" --org-id "024daf48ce22a000"

- The above command returns the following output (Name of the organization is missing):
 > ID			        Name		    Retention	Organization	OrganizationID
    024db19eb862a000	bucket-influx-4	1s				         024daf48ce22a000

- Run 'influx bucket find`. It will return an error:
> organization not found (error reference code: 1)

### LINKED ISSUE #14442: `influx create bucket` functionality does not match documentation
__Steps to reproduce:__

1. Attempt to create a new bucket on influxdb 2.x from the command line following the documentation: https://v2.docs.influxdata.com/v2.0/organizations/buckets/create-bucket/
2. Run the example given in the documentation `influx bucket create -n my-bucket -o my-org -r 604800000000000`

__Expected behavior:__
This is expected to create a new bucket "my-bucket" in org "my-org" with a retention period of 604800000000000 nanoseconds

__Actual behavior:__
The immediate error on running is that `--org-id` flag does not have the shorthand `-o` as described in the documentation, which results in this output `Error: unknown shorthand flag: 'o' in -o`. 

Changing `-o` to `--org-id` prevents that error, but exposes an issue with the `-r` flag. The documentation describes the retention period as a unit of nanoseconds, but using `-r 604800000000000` returns `Error: invalid argument "604800000000000" for "-r, --retention" flag: time: missing unit in duration 604800000000000`.

Changing the retention period to `-r 10s` prevents the error. However, this exposes yet another issue. The documentation describes the `--org-id` as "The name or ID of the organization to which it belongs", but only attempts to use the flag as an orgID, resulting in the message `Error: Failed to decode org id "my-org": <invalid> id must have a length of 16 bytes.`.

Changing the `--org-id` flag to a valid org influxdb.ID will finally allow you to create a bucket, however, as described here: https://github.com/influxdata/influxdb/issues/10716, the new bucket is never assigned an orgID.

__Environment info:__
Darwin 17.7.0 x86_64

### PR DESCRIPTION
**Pull request by zhulongcheng**
Closes #14442
Closes #10716

Describe your proposed changes here.

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [ ] Tests pass
- [ ] http/swagger.yml updated (if modified Go structs or API)
- [ ] Documentation updated or issue created (provide link to issue/pr)
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

**Comment by jademcgough** [comment]
alchemy please merge

File: cmd/influx/bucket.go [hunk 1]
```diff
@@ -58,8 +60,8 @@ func newBucketService(f Flags) (platform.BucketService, error) {
 }
 
 func bucketCreateF(cmd *cobra.Command, args []string) error {
-	if bucketCreateFlags.orgID == "" {
-		return fmt.Errorf("must specify org-id")
+	if bucketCreateFlags.orgID != "" && bucketCreateFlags.org != "" {
+		return fmt.Errorf("must specify at exactly one of org and org-id")
```
Comment (jademcgough) [inline cmd/influx/bucket.go]: change this to `must specify exactly one of org and org-id`

File: cmd/influx/bucket.go [hunk 2]
```diff
@@ -58,8 +60,8 @@ func newBucketService(f Flags) (platform.BucketService, error) {
 }
 
 func bucketCreateF(cmd *cobra.Command, args []string) error {
-	if bucketCreateFlags.orgID == "" {
-		return fmt.Errorf("must specify org-id")
+	if bucketCreateFlags.orgID != "" && bucketCreateFlags.org != "" {
```
Comment (jademcgough) [inline cmd/influx/bucket.go]: We should guard against the situation where neither flag is set.

Add another check for `if bucketCreateFlags.orgID == "" && bucketCreateFlags.org == ""`

**Comment by zhulongcheng** [comment]
Updated. Thanks so much @jademcgough

**Pull request closed without merging**